EHN: modify the example in README

tyrinwu opened the pull request:
I tried to make the example more easy to follow by rearranging the sequence that information is presented. 

First, let users know what the two files contain. Then, start to work on the annotation part.

mpage commented on README.rst:
```diff
@@ -8,17 +8,33 @@ directly to your Python code based on the types collected at runtime.
 
   with monkeytype.trace():
-      ...
+      ...a block of code in myscript.py
```
The `a block of code in myscript.py` comment is redundant. The previous sentence states that this code lives in `myscript.py`.

tyrinwu replied:
Thanks for the review. Updated.

carljm commented on README.rst:
```diff
@@ -8,17 +8,33 @@ directly to your Python code based on the types collected at runtime.
   $ monkeytype run myscript.py
 
-Or enable call-trace logging for a block of code::
+Or enable call-trace logging for a block of code in ``myscript.py``::
```
I'm slightly concerned that this gives the wrong impression that the context manager is only for use in a script. The main reason the context manager exists is that in some cases you might want to run tracing on some code that isn't a script at all. I think it might be better to move the context manager down to after the main example, letting the primary example flow more smoothly with only `monkeytype run`.

tyrinwu replied:
You mean that it can also be used in an interactive shell?

 I actually feel this example is a bit too long as well.

tyrinwu commented:
@carljm @mpage 
I committed a shorter version. I dropped the context manager part because 
   1. I am not sure how to integrate it without cutting the flow of the document.
   2. It seems alright not to talk about it in the beginning.

I moved the limitation part to almost the end since it seems too serious to talk about it at the very beginning.

Sorry for the messy multiple commits.
